### DISCUSSION TITLE
[Question]: How Docker deployment hot-updates services when modifying code?

### DISCUSSION THREAD

**Discussion by Kevin-free** [danny-avila/LibreChat#2746] [Q&A] [answered]
### What is your question?

Hello, I am new to docker. How Docker deployment hot-updates services when modifying code?

I checked online, but my attempts didn't work:

```
# Rebuild the image
docker-compose build
# Restart the service
docker-compose up -d
```

Rebuilding the image works, but is time-consuming:
```
# Rebuild and push the image
docker build -t ghcr.io/NAMESPACE/NEW_IMAGE_NAME:latest .
docker push ghcr.io/NAMESPACE/NEW_IMAGE_NAME:latest
# Restart the service
docker-compose up -d
```

Is there any convenient method? Or how do you carry out local development, debugging and deployment? Thanks!

### More Details

Local environment: Mac 12.6.3

```
> docker version
Client:
  Cloud integration: v1.0.33
  Version: 24.0.2
  API version: 1.43
  Go version: go1.20.4
  Git commit: cb74dfc
  Built: Thu May 25 21:51:16 2023
  OS/Arch: darwin/arm64
  Context: desktop-linux

Server: Docker Desktop 4.20.1 (110738)
  Engine:
   Version: 24.0.2
   API version: 1.43 (minimum version 1.12)
   Go version: go1.20.4
   Git commit: 659604f
   Built: Thu May 25 21:50:59 2023
   OS/Arch: linux/arm64
   Experimental: false
  containerd:
   Version: 1.6.21
   GitCommit: 3dce8eb055cbb6872793272b4f20ed16117344f8
  runc:
   Version: 1.1.7
   GitCommit: v1.1.7-0-g860f061
  docker-init:
   Version: 0.19.0
   GitCommit: de40ad0
```

```
> docker-compose version
Docker Compose version v2.18.1
```

### What is the main subject of your question?

Installation, Other

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Kevin-free** [comment]
Including modifying the front-end `client` and back-end `api`

**Comment by danny-avila** [comment, marked as answer]
For development, it’s not recommended to use docker, see the guides here: https://www.librechat.ai/docs/development

**Reply by Kevin-free** [reply]
Sorry, I didn't read the document carefully. It turns out that you have already written it. This is a great project and a very complete document. Thank you!
